include/dgl/aten/csr.h

```diff
@@ -151,13 +162,17 @@ struct CSRMatrix {
    *       The context check is deferred to pinning the NDArray.
    */
   inline void PinMemory_() {
-    if (is_pinned) return;
-    indptr.PinMemory_();
-    indices.PinMemory_();
-    if (!aten::IsNullArray(data)) {
-      data.PinMemory_();
+    if (!IsEmpty()) {
+      if (is_pinned) return;
+      indptr.PinMemory_();
+      indices.PinMemory_();
+      if (!aten::IsNullArray(data)) {
+        data.PinMemory_();
+      }
+      is_pinned = true;
     }
     is_pinned = true;
```

Comment:
And here.

Reply:
This is probably necessary I think, as it is an in-place method. We should flag it even if it is empty when we called this function.